packages/pds/src/api/com/atproto/account.ts

```diff
@@ -52,17 +42,17 @@ export default function (server: Server) {
       )
     }
 
-    let isTestUser = false
-    if (username.endsWith('.test')) {
-      if (!config.debugMode || !config.testNameRegistry) {
-        throw new InvalidRequestError(
-          'Cannot register a test user if debug mode is not enabled',
-        )
-      }
-      isTestUser = true
+    const supportedUsername = config.availableUserDomains.some((host) =>
+      username.toLowerCase().endsWith(host),
```

Comment:
```suggestion
      username.toLowerCase().endsWith(`.${host}`),
```

Reply:
Oops, ignore this ☝️ I see the hosts already start with a dot.

Perhaps the alternate fix is just to validate that in `ServerConfig`.  The spirit of the suggestion is that we don't want to end-up in a situation where accounts are only allowed from `mydomain.com`, and someone is able to sign-up with a `notmydomain.com` account.